library/ssl_msg.c

```diff
@@ -5610,7 +5607,7 @@ void mbedtls_ssl_write_version( int major, int minor, int transport,
 #if defined(MBEDTLS_SSL_PROTO_DTLS)
     if( transport == MBEDTLS_SSL_TRANSPORT_DATAGRAM )
     {
-        if( minor == MBEDTLS_SSL_MINOR_VERSION_2 )
+        if( minor == MBEDTLS_SSL_MINOR_VERSION_3 - 1 )
             --minor; /* DTLS 1.0 stored as TLS 1.1 internally */
```

Comment:
This is still dead code, you're just hiding the MBEDTLS_SSL_MINOR_VERSION_2 as "3-1".
Remove whole blocks of code which are enabled only for removed version.

Reply:
See my comment right below.